pkg/actuators/machine/reconciler.go

```diff
@@ -64,7 +67,14 @@ func (r *Reconciler) create() error {
 		return fmt.Errorf("failed to get user data: %w", err)
 	}
 
-	instance, err := launchInstance(r.machine, r.providerSpec, userData, r.awsClient)
+	infra := &configv1.Infrastructure{}
+	infraName := client.ObjectKey{Name: awsclient.GlobalInfrastuctureName}
+
+	if err := r.client.Get(r.Context, infraName, infra); err != nil {
+		return err
```

Comment:
Please add some context to this error and add a log line please

Do we want to fail if there is no infra object? ie should we be accounting for a not found error at all?

Reply:
I think that infra object should always be present.